### PR TITLE
Custom redshift language

### LINKED ISSUE #153: Define and load custom redshift sql language

### PR DESCRIPTION
**Pull request by sunker**
This PR composes and uses a custom redshift Monaco language that is based on the standard sql language from grafana experimental. This was made possible in [this](https://github.com/grafana/grafana-experimental/pull/35) this PR in grafana experimental. 
 
Part of https://github.com/grafana/redshift-datasource/issues/108
Fixes #153

### CODE REVIEW

File: src/language/language.ts [hunk 1]
```diff
@@ -0,0 +1,661 @@
+import { SQLMonarchLanguage, grafanaStandardSQLLanguage, grafanaStandardSQLLanguageConf } from '@grafana/experimental';
+
+// Language based on standard SQL, but edited according to https://docs.aws.amazon.com/redshift/latest/dg/c_SQL_commands.html
+
+export const logicalOperators = [
```
Comment (andresmgot) [inline src/language/language.ts:5]: all this is already written in the official language for Redshift: https://github.com/microsoft/monaco-editor/blob/main/src/basic-languages/redshift/redshift.ts#L34

can we use it here?

**Review by andresmgot — COMMENTED** [review]
My main question here is that if we are able to extend the existing monaco language for redshift so we don't need to re-define everything?

**Comment by sunker** [comment]
> My main question here is that if we are able to extend the existing monaco language for redshift so we don't need to re-define everything?

Any language from the Monaco language registry can be used, but extending it is not possible. But we don't want to use the redshift language from the Monaco registry for a few reasons:
* It's not official or actively maintained. Comparing for example the functions in the [official language ref](https://docs.aws.amazon.com/redshift/latest/dg/c_SQL_functions.html), there's a bunch of things missing in the Monaco language
* Composing a new language based on the standard [grafana sql language](https://github.com/grafana/grafana-experimental/blob/ec1694d3b1f023adf2a842c809affeb1cdafc8b5/src/sql-editor/standardSql/language.ts#L57) will give syntax highlighting for template variables and macros
* Differentiating different operators using the [SQLMonarchLanguage](https://github.com/grafana/grafana-experimental/blob/ec1694d3b1f023adf2a842c809affeb1cdafc8b5/src/sql-editor/standardSql/types.ts#L42-L43) will enable a better autocomplete experience 

Plan to update the redshift language definition so that it's at least somewhat close to [official language ref](https://docs.aws.amazon.com/redshift/latest/dg/c_SQL_functions.html) in a future PR

**Comment by sunker** [comment]
@andresmgot any idea of what's causing the build error?

**Pull request merged**